frontend/app/ProductInfo.tsx

```diff
@@ -9,29 +9,141 @@ import {
   TextInput,
 } from "react-native";
 import { SafeAreaView } from "react-native-safe-area-context";
-import ProdPageHeader from "@/components/add-product/AddProductHeader";
 import { Theme } from "@/constants/theme";
 import { useLocalSearchParams } from "expo-router";
 import AddProductHeader from "@/components/add-product/AddProductHeader";
 import { useRouter } from "expo-router";
 import { useProductStore } from "@/store/productStore";
 import { useState } from "react";
+import { accessToken, MOBILE_API_URL } from "@/utils/authUtils";
+import { Picker } from "@react-native-picker/picker";
 
 export default function ProductInfo() {
   const productData = useLocalSearchParams();
   const { addProduct } = useProductStore();
   const router = useRouter();
   const [name, setName] = useState("");
   const [price, setPrice] = useState("");
+  const [condition, setCondition] = useState("Brand New");
 
-  const handlePublish = () => {
-    addProduct({
-      id: Date.now().toString(),
-      title: name,
-      price: Number(price),
-      imageUri: productData?.imageUri as string,
+  const apiUrl = MOBILE_API_URL;
+  const getSignedUrl = async (filename: string) => {
+    const res = await fetch(`${apiUrl}/upload-url`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`,
+      },
+      body: JSON.stringify({
+        images: [
+          { filename, mimeType: "image/png" },
+          { filename, mimeType: "image/jpeg" },
+          { filename, mimeType: "image/jpg" },
+        ],
+      }),
     });
-    router.back();
+    const data = await res.json();
+    return data[0];
+  };
+
+  const uploadToSignedUrl = async (signedUrl: string, imageUri: string) => {
+    const image = await fetch(imageUri);
+    const blob = await image.blob();
+
+    const upload = await fetch(signedUrl, {
+      method: "PUT",
+      headers: {
+        "Content-Type": blob.type || "application/octet-stream",
+      },
+      body: blob,
+    });
+
+    if (!upload.ok) throw new Error("Upload failed");
+  };
+
+  const createListing = async (filePath: string) => {
+    const mutation = `
```

Comment:
lagay mo to sa separate utils.
`/graphql/mutations.ts`
andun lahat ng mutation mo

```suggestion
export const CreateListMutation = `
      mutation createListing($data: CreateListingInput!) {
        createListing(data: $data) {
          message
        }
      }
    `;
`
```